src/hooks/useActions.js

```diff
@@ -216,43 +223,68 @@ export default function useActions() {
     [account, agreementApp, mounted]
   )
 
-  const approveChallengeTokenAmount = useCallback(
-    (depositAmount, onDone = noop) => {
-      if (!feeTokenContract || !agreementApp) {
+  const approveTokenAmount = useCallback(
+    (amount, tokenContract, appAddress, trxDescription) => {
+      if (!tokenContract || !appAddress) {
         return
       }
-      const approveData = encodeFunctionData(feeTokenContract, 'approve', [
-        agreementApp.address,
-        depositAmount.toString(10),
+      const approveData = encodeFunctionData(tokenContract, 'approve', [
+        appAddress,
+        amount.toString(10),
       ])
       const intent = [
         {
           data: approveData,
           from: account,
-          to: feeTokenContract.address,
-          description: 'Approve HNY',
+          to: tokenContract.address,
+          description: trxDescription,
         },
       ]
 
+      return intent
+    },
+    [account]
+  )
+
+  const approveChallengeTokenAmount = useCallback(
+    (depositAmount, onDone = noop) => {
+      if (!feeTokenContract || !agreementApp) {
+        return
+      }
+
+      const intent = approveTokenAmount(
+        depositAmount,
+        feeTokenContract,
+        agreementApp.address,
+        'Approve HNY'
+      )
+
       if (mounted()) {
         onDone(intent)
       }
     },
-    [account, feeTokenContract, agreementApp, mounted]
+    [approveTokenAmount, feeTokenContract, agreementApp, mounted]
   )
 
-  const getAllowance = useCallback(async () => {
-    if (!feeTokenContract) {
-      return
-    }
+  const getAllowance = useCallback(
+    async (tokenContract, appAddress) => {
+      if (!tokenContract) {
+        return
+      }
 
-    const allowance = await feeTokenContract.allowance(
-      account,
-      agreementApp.address
-    )
+      const allowance = await tokenContract.allowance(account, appAddress)
 
-    return new BigNumber(allowance.toString())
-  }, [account, feeTokenContract, agreementApp])
+      return new BigNumber(allowance.toString())
+    },
+    [account]
+  )
+
+  const getAgreementAllowance = useCallback(async () => {
+    if (!agreementApp || !feeTokenContract) {
+      return
+    }
+    return await getAllowance(feeTokenContract, agreementApp.address)
+  }, [agreementApp, feeTokenContract, getAllowance])
```

Comment:
Since getAllowance is an async function there's no need to do an async await here
```suggestion
  const getAgreementAllowance = useCallback(() => {
    if (!agreementApp || !feeTokenContract) {
      return
    }
    return getAllowance(feeTokenContract, agreementApp.address)
  }, [agreementApp, feeTokenContract, getAllowance])
```